javascript/extractor/src/com/semmle/js/extractor/RegExpExtractor.java

```diff
@@ -122,17 +121,14 @@ private Label extractTerm(RegExpTerm term, Label parent, int idx) {
   }
 
   public void emitLocation(SourceElement term, Label lbl) {
-    int col = literalStart.getColumn();
-    int sl, sc, el, ec;
-    sl = el = literalStart.getLine();
-    sc = col + offsets.get(term.getLoc().getStart().getColumn());
-    ec = col + offsets.get(term.getLoc().getEnd().getColumn());
-    sc += 1; // convert to 1-based
-    ec += 1; // convert to 1-based
-    ec -= 1; // convert to inclusive
+    int sl = sourceMap.getStart(term.getLoc().getStart().getColumn()).getLine();
+    int sc = sourceMap.getStart(term.getLoc().getStart().getColumn()).getColumn() + 1; // convert to 1-based
+    int el = sourceMap.getEnd(term.getLoc().getEnd().getColumn()).getLine();
+    int ec = sourceMap.getEnd(term.getLoc().getEnd().getColumn()).getColumn() - 1; // convert to inclusive
```

Comment:
Is this right? The `ec` modifications used to be a noop:
```
    ec += 1; // convert to 1-based
    ec -= 1; // convert to inclusive
```
I can't quite see if the use of `sourceMap` makes it correct.

Reply:
It wasn't right, there was some off-by-one errors.   

I'm not sure why that is, but there is some conversion between 0-based and 1-based columns, so that seems to be some of it.  

I've fiddled around with it, and now I got something that works (but I'm not quite sure why it works).  

I've manually checked all the locations emitted for `RegExpTerms` in [`multipart.js`](https://github.com/github/codeql/pull/6756/files#diff-dda2e541a3b7330aded21f23ed3cfece939be427d0a2f1bac40f095ff61d899f).  
(I did that by running a test-query in VSCode, clicking results, and checking that the highlights were correct).